flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java

```diff
@@ -47,6 +47,11 @@
 			"Class with the program entry point (\"main\" method or \"getPlan()\" method. Only needed if the " +
 					"JAR file does not specify the class in its manifest.");
 
+	static final Option CLASSPATH_OPTION = new Option("C", "classpath", true, "Adds a URL to each user code classloader " +
```

Comment:
HDFS is not possible yet. the user would have to provide a special URL protocol handler via `java -Djava.protocol.handler.pkgs=org.my.protocols` such that the URLClassloader understands the protocol.